pkgs/tools/system/hwinfo/default.nix

```
@@ -37,15 +42,27 @@ stdenv.mkDerivation rec {
     substituteInPlace Makefile --replace "/sbin" "/bin" --replace "/usr/" "/"
     substituteInPlace src/isdn/cdb/Makefile --replace "lex isdn_cdb.lex" "flex isdn_cdb.lex"
     substituteInPlace hwinfo.pc.in --replace "prefix=/usr" "prefix=$out"
+
+    # replace absolute paths with relative, we will prefix PATH later
+    substituteInPlace src/hd/hd_int.h \
+      --replace-fail "/sbin/" "" --replace-fail "/usr/bin/" ""
```

It might be nice to avoid `wrapProgram` and adding `PATH` and instead replace `/sbin/modprobe` with `${kmod}/bin/modprobe`, etc.

```suggestion
    # replace absolute paths with relative, we will prefix PATH later
    substituteInPlace src/hd/hd_int.h \
      --replace-fail "/sbin/modprobe" "${kmod}/bin/modprobe" \
      --replace-fail "/sbin/rmmod" "${kmod}/bin/rmmod" \
      --replace-fail "/usr/bin/udevinfo" "${systemdMinimal}/bin/udevinfo" \
      --replace-fail "/usr/bin/udevadm" "${systemdMinimal}/bin/udevadm"

    # check for leftover references to FHS binaries.
    if grep /sbin /usr/bin src/hd/hd_int.h
    then
      echo "Santity check failed. The above lines should be replaced in src/hd/hd_int.h"
      exit 1
    fi
```